### PR TITLE
feat(sanity): add Media Library import and export CLI commands 

### PR DESCRIPTION
**Pull request by juice49**
### Description

This branch adds `media import` and `media export` commands to the Sanity CLI. These commands can be use to export and import media library assets.

Media Library archives use a similar format to dataset archives, with a few key differences:

- The `data.ndjson` file:
  - Includes only `sanity.asset` document types.
  - Includes only the asset filename and aspect data.
  - Always excludes drafts.
  - Omits asset documents that have no aspect data.
- The assets map (`assets.json`) is omitted (the presence of assets in the archive is instead considered the source of truth).

When importing a media library archive:

- The importer uses the contents of the `files` and `images` directories as the list of assets to upload. The importer first checks if any of these assets have already been uploaded (based on their SHA-1 hash), and then uploads those that have not.
- After uploading an asset, the importer searches for a matching filename in `data.ndjson` and performs a mutation to set any aspect data found.
- The SHA-1 hash of the asset acts as the asset id. If an existing asset document currently references an asset instance matching this hash as its source, aspects from the imported archive will be written to it. **Aspect data is not merged:** the aspects object from `data.ndjson` overwrites existing aspect data.
- Existing aspect data will only be overwritten if the importer is run with the `--replace-aspects` option switched on. If this option is not switched on, aspect data will only be set for new assets.

#### Future improvements

- More granular upload progress.
- The ability to control filenames.
- The ability to import particular aspect data (rather than setting the entire aspects object).

### What to review

- The import and export commands. Particularly the import command, which introduces the most new code.

### Testing

- Added unit tests.
- Performed many test imports to a production media library.

### CODE REVIEW

File: packages/sanity/package.json
Comment (jordanl17) [inline packages/sanity/package.json:1]: Q: Was there any review done on these new dependencies? Do we use them elsewhere, or are there any alternatives that aren't suitable? Just want to understand the lift we will be taking on in maintaining them going forward (eg I see `peek-stream` hasn't been updated for 7 years)

Comment (juice49) [reply packages/sanity/package.json:1]: Yep—the tar handling is lifted almost directly from `@sanity/export` where they have lived happily for a long time. I've added no dependencies that aren't already in use there.

File: packages/sanity/src/_internal/cli/actions/media/importAssetsAction.ts [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+    scan<State, [number, State | undefined]>(
+      (processedAssetsCount, state) => [processedAssetsCount[0] + 1, state],
+      [0, undefined],
+    ),
+    tap({
+      next: ([processedAssetsCount, state]) => {
+        previousState = state
+        spinner.text = `${processedAssetsCount} of ${state?.fileCount} assets imported ${chalk.dim(state?.asset.originalFilename)}`
+      },
+      complete: () => spinner.succeed(`Imported ${previousState?.fileCount} assets`),
```
Comment (jordanl17) [inline packages/sanity/src/_internal/cli/actions/media/importAssetsAction.ts:488]: COULD: Could be nice to track assets that are skipped because they exist, assets that failed cause of errors

File: packages/sanity/src/_internal/cli/actions/media/importAssetsAction.test.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
+import path from 'node:path'
+
+import chalk from 'chalk'
+import {lastValueFrom, of, toArray} from 'rxjs'
+import {describe, expect, it} from 'vitest'
+
+import {type AssetWithAspects, resolveSource, setAspects} from './importAssetsAction'
+import {createMockClient} from './test/createMockClient'
+
+describe('resolveSource', () => {
```
Comment (jordanl17) [inline packages/sanity/src/_internal/cli/actions/media/importAssetsAction.test.ts:10]: Could: 
Add in a test for a missing or empty `data.ndjson`
Test that `assetIds: []` is handled without throw

**Review by jordanl17 — DISMISSED** [review]
Only some minor suggestions - happy for you to take or leave them; or split them into a separate PR

File: packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      // The `assets.json` assets map is not required for Media Library archives, because the
+      // import process reads asset files directly from the archive.
+      assetsMap: false,
+      // The documents listed in `data.ndjson` are only used for recording aspect data. If there
+      // is no aspect data, the document can safely be omitted.
+      filterDocument: (doc: unknown) => {
+        if (typeof doc !== 'object' || doc === null || !('aspects' in doc)) {
+          return false
+        }
+        return Array.isArray(doc.aspects) && Object.keys(doc.aspects).length !== 0
```
Comment (sgulseth) [inline packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts:67]: aspects are an object, so this will always be false.

The second clause is probably what you want? 
```suggestion
        return Object.keys(doc.aspects).length !== 0
```

**Review by sgulseth — CHANGES_REQUESTED** [review]
Looks good, we just need to fix the aspect check!

File: packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts (hunk 3 above)
Comment (juice49) [reply packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts]: Fixed this now. Thanks also @jordanl17 for noticing there was a problem here.

[the remark this answers is not in the document]
Comment (juice49) [reply packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts]: Thanks for spotting!

File: packages/sanity/src/_internal/cli/actions/media/importAssetsAction.ts (hunk 1 above)
Comment (juice49) [reply packages/sanity/src/_internal/cli/actions/media/importAssetsAction.ts:488]: Yep, I agree! For now, running the command in debug mode will provide explicit logging in case users need to debug issues.

File: packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  spinner.succeed(`Exported media library to ${chalk.bold(outputPath)}`)
+}
+
+export default exportAssetsAction
+
+function generateFilename(assetId: string): string {
+  const [, , asset, ext] = assetId.match(/^(image|file)-(.*?)(-[a-z]+)?$/) || []
+  const extension = (ext || 'bin').replace(/^-/, '')
+  return asset ? `${asset}.${extension}` : `${assetId}.bin`
```
Comment (sgulseth) [inline packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts:116]: I didn't see this in earlier review, but we have tooling for these kind of operations already here: https://github.com/sanity-io/asset-utils#buildfilepath

Though we can fix in a followup

File: packages/sanity/src/_internal/cli/actions/media/importAssetsAction.ts [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+  const mediaLibraryId =
+    args.extOptions['media-library-id'] ?? (await determineTargetMediaLibrary(context))
+
+  const client = apiClient().withConfig({
+    'apiVersion': MINIMUM_API_VERSION,
+    'requestTagPrefix': 'sanity.mediaLibraryCli.import',
+    '~experimental_resource': {
+      type: 'media-library',
+      id: mediaLibraryId,
+    },
```
Comment (sgulseth) [inline packages/sanity/src/_internal/cli/actions/media/importAssetsAction.ts:134]: :chefs-kiss:

**Review by sgulseth — APPROVED** [review]
nice! one nit about tooling, but fore the sake of keeping PRs small and short lived we can merge and fix in a follow-up!

File: packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts (hunk 4 above)
Comment (juice49) [reply packages/sanity/src/_internal/cli/actions/media/exportAssetsAction.ts:116]: Ah, thanks! I didn't spot that function in the docs when I checked. We should definitely adopt it in a follow-up.

**Review by sgulseth — APPROVED** [review]
